### ISSUE TITLE
Wind thresholds changed recently

### ISSUE THREAD

**Issue by rawrbooks** [5310/discord-bot-castform#20]
Hi,

I and others in my discord server have noticed that the game has changed how it accounts for gusts -- either pulling wind based solely on gust speed or something else about how it relates to overall wind speed. Regardless, I'm now getting lots of "phantom wind", wind that shows up in-game but not in the bot. I'm down to do some documentation in order to figure out exactly what has shifted, if y'all could change the code so that it uses a more accurate variable. Could someone partner with me for it?

**Comment by 5310** [comment]
Hullo there! Let me make a branch for you for testing and observations: [v2-webhook-rawrbooks](https://github.com/5310/discord-bot-castform/tree/v2-webhook-rawrbooks)

Most of 2021 I myself got a lot of unexpected wind, and it was pretty accurate, but then winter happened (when we normally get a lot of wind IRL) and neither the game nor the bot has said it'd be windy. It's very rare here, so I'm glad you're looking into it again.

I'll give you edit access to the branch, but you an also tell me what changes to the current thresholds or formula you want to make to start with and I'll edit it in.

**Comment by rawrbooks** [comment]
hi!! thank you! I actually am in the very beginning stages of learning to code, so not sure that I'd be able to help with edits, but I can certainly track my weather and in-game responses to try and get an idea of where the pull formula might have changed. I'll tap a few friends to try and come up with hypotheses of what's happening too! I'll make a note to follow up in a couple weeks and see what data I can gather in the meantime, but I had some phantom wind yesterday where it was supposed to be partly cloudy and was windy instead lol.

**Comment by ajstewart** [comment]
I'd be happy to also look over any data you get. It's always the nicer side of the problem if wind is happening in-game but not predicted by the current model!

I am not much help at the moment 😅:
<img width="299" alt="pogo-weather" src="https://user-images.githubusercontent.com/3306161/152993814-a2f51472-c11c-4f53-a9a4-df911b6709ba.png">

**Comment by rawrbooks** [comment]
Hi!

I have been a little delinquent with gathering data but will start doing so
again this week and next week.

Do you know how to calculate local pull times (like the in-game ones rather
than the ones we set with the bot)? I'm still not totally clear and
sometimes the time period when the weather matches the bot's predictions
varies. I can send an example if that would help clarify my question!

**Comment by 5310** [comment]
Honestly, I gave up trying to estimate the pull times completely...

In my region I've never noticed the pulls update more than once in an hour on the hour. But AJ has observed them changing within the hour more than once, and so much of it is still in the air :smiling_face_with_tear: 

In my case, the updates happened anything from 10 to 15 minutes after the hour. Which I found out by querying every five minutes for an hour, and noting when they actually updated. Since they update only once, even that fuziness of _when_ the data actually shows up in a query wasn't enough to worry about. I just let the bot pull in the middle of the hour.

**Comment by rawrbooks** [comment]
Hahaha yeah it seems like the hour of the day that they happen changes here
too (I have a notes section in my server and it's littered with weather
observations where I've been noting when the weather predictions are
accurate (assuming there's one hour where they are after that, I've been
assuming that's the pull time and others are 8h after and 16 after
respectively)). :'(

My friend who also noticed the wind changes theorized that it might have
had to do with gust vs constant wind speed. I've had wind in game today
almost the whole evening, with none of it predicted by castform. If I
recall correctly, AccuWeather had a constant wind speed of under 25km/h but
gusts of 40km/h, so maybe they're going by gust speed instead of constant
wind now? Or maybe the threshold was lowered 🤔 I'll try to gather some
more observations RE predicted constant speed vs gusts.

Hannah Madsen, MA, MBA, MLIS
Email: ***@***.***
Cell: 780-940-8765
LinkedIn: hbmadsen

**Comment by ajstewart** [comment]
Yes I have done an experiment before where I have observed the wind speed predictions changing within the hour that has the possibility to affect the forecast, but I'd be surprised if this was a consistent problem. I gave up a while ago now trying to predict the actual pull time, I notice now and then that the forecast still lags behind by a number of hours (rain is very helpful with that as it overrides the wind). But that combined with the minute of the pull, the uncertainty that we are actually using the correct Accuweather location etc etc makes it pretty painful to try and nail down. I settled on `:35` for the pull and left it at that.

It's interesting that you have no wind prediction at all though and wind in game. I haven't monitored much myself in recent times but I always had the problem the other way around - predicting wind when it never came. There was always a feeling that there was an upper gust limit that would trigger it but I was never able to see this in my own data conclusively.

But yes this was all a while ago now so it absolutely could have changed!

**Comment by rawrbooks** [comment]
I have phantom wind today (I'm based out of Edmonton Alberta, and it's both
flat and gets airflow from polar vortexes etc, so we get loads of wind all
the time), so I'm doing some hourly tracking. do we know what the current
bot formula is for constant vs gust speed that triggers a wind prediction?

H

**Comment by ajstewart** [comment]
The relevant logic is here:
https://github.com/5310/discord-bot-castform/blob/adcef34a015466c1f22a5b1bd97d20a8561db742/model-ajstewart.js#L155-L182
Line 173 is the windy decision.

However on my instance I run the gust threshold as `29.6`, not `30` as is used here in line 158.

**Comment by rawrbooks** [comment]
interesting! a friend coded it to my server for me, so I've asked him for
the code he used (just to double check the wind thresholds are the same).

Here's what I got so far today:

[image: image.png]

As silly as it is, I wonder if they lowered the constant wind speed to 19
or 18 km/h and that's what's creating the issue. I'm going to keep tracking
and see if I get the same thing again, assuming that I have more days with
medium wind instead of crazy wind.

Cheers,

**Comment by FuzionEcho** [comment]
Hello!, so I've been trying to run this on my discord server, but I can't quite seem to be able to grasp how to is there any way you can make a video tutorial on how to do it?
